lib/ansible/plugins/inventory/ovirt.py

```diff
@@ -0,0 +1,275 @@
+        '''
+        return [self._get_dict_of_struct(host) for host in self._get_hosts(query_filter=query_filter)]
+
+    def _get_hosts(self, query_filter=None):
+        '''
+            :param filter: dictionary of vm filter parameter/values
+            :return list of oVirt vm structs
+        '''
+        connection = self._get_connection()
+
+        vms_service = connection.system_service().vms_service()
+        if query_filter is not None:
+            return vms_service.list(**query_filter)
+        else:
+            return vms_service.list()
+
+    def _get_query_options(self, param_dict):
+        ''' Get filter parameters and cast these to comply with sdk VmsService.list param types
+            :param param_dict: dictionary of filter parameters and values
+            :return dictionary with casted parameter/value
+        '''
+        if param_dict is None:
+            return None
+
+        FILTER_MAPPING = {
+            'all_content': bool,
+            'case_sensitive': bool,
+            'filter': bool,
+            'follow': str,
+            'max': int,
+            'search': str
+        }
+
+        casted_dict = {}
+
+        for (param, value) in param_dict.items():
+            try:
+                casted_dict[param] = FILTER_MAPPING[param](value)
+            except KeyError:
+                raise AnsibleError("Unknown filter option '{0}'".format(param))
+
+        return casted_dict
+
+    def _get_hostname(self, host):
+        '''
+          Get the host's hostname based on prefered attribute
+          :param host: dict representation of oVirt VmStruct
+          :param return: preferred hostname for the host
+        '''
+        hostname_preference = self.get_option('ovirt_hostname_preference')
+        if not hostname_preference:
+            raise AnsibleParserError('Invalid value for option ovirt_hostname_preference: {0}'.format(hostname_preference))
+        hostname = None
+
+        for preference in hostname_preference:
+            hostname = host.get(preference)
+            if hostname is not None:
+                return hostname
+
+        raise AnsibleParserError("No valid name found for host id={0}".format(host.get('id')))
+
+    def _populate_from_source(self, source_data):
+
+        for host in source_data:
+
+            hostname = self._get_hostname(host)
+
+            self.inventory.add_host(hostname)
+
+            for fact, value in host.items():
+                self.inventory.set_variable(hostname, fact, value)
+
+            self._set_composite_vars(self.get_option('compose'), host, hostname)
+            self._add_host_to_composed_groups(self.get_option('groups'), host, hostname)
+            self._add_host_to_keyed_groups(self.get_option('keyed_groups'), host, hostname)
+
+    def verify_file(self, path):
+
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('ovirt.yml', 'ovirt4.yml', 'ovirt.yaml', 'ovirt4.yaml')):
```

Comment:
I'm not sure with logic that the plugin would accept only files which ends with `ovirt.yml` etc.
What if it would accept any file?

Reply:
The convention across inventory plugins is that the file is in the format `*<NAME>.yml|yaml`. It is only really governed by this method but this is consistent with other plugins.

Reply:
okay if its standard keep it this way